test/libsolidity/semanticTests/various/address_code.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    bytes public initCode;
+
+    constructor() {
+        // This should catch problems, but lets also test the case the optimiser is buggy.
+        assert(address(this).code.length == 0);
+        initCode = address(this).code;
+    }
+
+    function f() public view returns (uint) { return address(this).code.length; }
```

Comment:
Not sure if we don't want to rely on exact bytecode length here, if not, we could replace this with something like `address(this).code.length > 100`.